azurerm/resource_arm_cosmos_mongo_database.go

```diff
@@ -0,0 +1,153 @@
+	client := meta.(*ArmClient).cosmosAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	account := d.Get("account_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.GetMongoDBDatabase(ctx, resourceGroup, account, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of creating Cosmos Mongo Database %s (Account %s): %+v", name, account, err)
+			}
+		} else {
+			id, err := azure.CosmosGetIDFromResponse(existing.Response)
+			if err != nil {
+				return fmt.Errorf("Error generating import ID for Cosmos Mongo Database '%s' (Account %s)", name, account)
+			}
+
+			return tf.ImportAsExistsError("azurerm_cosmos_mongo_database", id)
+		}
+	}
+
+	db := documentdb.MongoDBDatabaseCreateUpdateParameters{
+		MongoDBDatabaseCreateUpdateProperties: &documentdb.MongoDBDatabaseCreateUpdateProperties{
+			Resource: &documentdb.MongoDBDatabaseResource{
+				ID: &name,
+			},
+			Options: map[string]*string{},
+		},
+	}
+
+	future, err := client.CreateUpdateMongoDBDatabase(ctx, resourceGroup, account, name, db)
+	if err != nil {
+		return fmt.Errorf("Error issuing create/update request for Cosmos Mongo Database %s (Account %s): %+v", name, account, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting on create/update future for Cosmos Mongo Database %s (Account %s): %+v", name, account, err)
+	}
+
+	resp, err := client.GetMongoDBDatabase(ctx, resourceGroup, account, name)
+	if err != nil {
+		return fmt.Errorf("Error making get request for Cosmos Mongo Database %s (Account %s): %+v", name, account, err)
+	}
+
+	id, err := azure.CosmosGetIDFromResponse(resp.Response)
+	if err != nil {
+		return fmt.Errorf("Error creating Cosmos Mongo Database '%s' (Account %s) ID: %v", name, account, err)
+	}
+	d.SetId(id)
+
+	return resourceArmCosmosMongoDatabaseRead(d, meta)
+}
+
+func resourceArmCosmosMongoDatabaseRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cosmosAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseCosmosDatabaseID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.GetMongoDBDatabase(ctx, id.ResourceGroup, id.Account, id.Database)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Error reading Cosmos Mongo Database %s (Account %s) - removing from state", id.Database, id.Account)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Cosmos Mongo Database %s (Account %s): %+v", id.Database, id.Account, err)
+	}
+
+	if props := resp.MongoDBDatabaseProperties; props != nil {
+		d.Set("name", props.ID)
+		d.Set("resource_group_name", id.ResourceGroup)
+		d.Set("account_name", id.Account)
```

Comment:
since these both come from the `id` field rather than the `props` block, these can be outside of this block